subscriber.go

```diff
@@ -649,6 +688,24 @@ func (s *Subscriber) Announce(ctx context.Context, nextCid cid.Cid, peerID peer.
 	return nil
 }
 
+func (s *Subscriber) republish(ctx context.Context, nextCid cid.Cid, peerID peer.ID, addrs []multiaddr.Multiaddr) error {
+	pidStr := peerID.String()
+	buf := bytes.NewBuffer(nil)
+	buf.Grow(len("republish") + len(pidStr))
+	buf.WriteString("republish")
```

Comment:
can we not add this to the ipld schema of the msg?

Reply:
That would be best, but I need a way to tell the cbor codec that the field is optional so that messages from lotus, that do not have it, do not appear [invalid](https://github.com/filecoin-project/go-legs/blob/main/dtsync/cbor_gen.go#L91-L93) because they are missing a field.  Is there some tag like
```go
`cbor:",omitempty"`
```
?

Reply:
I wrote the cbor_gen.go so that it is backward compatible with the old messages, and detects whether or not the new field is present.  We can leave this custom codec in place until lotus is using the new message format.